package.json

```diff
@@ -1,7 +1,16 @@
 {
+  "private": true,
   "devDependencies": {
     "@commitlint/cli": "^6.1.3",
-    "@commitlint/config-conventional": "^6.1.3"
+    "@commitlint/config-conventional": "^6.1.3",
+    "husky": "^2.4.1"
   },
-  "dependencies": {}
+  "dependencies": {},
+  "husky": {
+    "hooks": {
+      "pre-commit": "cd app/ui-react; yarn precommit",
+      "pre-push": "BRANCH=$(git rev-parse --abbrev-ref HEAD) UICHANGES=$(git diff --name-only --diff-filter=b $(git merge-base HEAD@{1} $BRANCH) | grep app/ui-react); if [[ -z \"${UICHANGES}\" ]]; then echo 'No changes on ui-react detected'; else cd app/ui-react && yarn prepush; fi"
```

Comment:
Now that I think about this, I'm not sure it will work if the latest commit didn't change something in the ui-react folder. Any idea?

Reply:
This is what I did for the incremental build thinggy: https://github.com/syndesisio/syndesis/blob/master/tools/bin/commands/util/common_funcs#L153

Reply:
Ok so I changed the script to use `syndesis -m ui-react --incremental` instead, and also added `npm-run-all` to pave the way to future additional hooks. Just add a new `[hook]:[module]` script in the `scripts` section with the script that needs to be run and it will be executed automatically.